nixos/modules/services/web-apps/firefly-iii.nix

```diff
@@ -0,0 +1,387 @@
+{ pkgs, config, lib, ... }:
+
+let
+  inherit (lib) mkDefault mkIf mkOption mkPackageOption mkEnableOption literalExpression mdDoc;
+  inherit (lib.types) attrsOf oneOf str int bool path package enum submodule;
+
+  cfg = config.services.firefly-iii;
+
+  user = cfg.user;
+  group = cfg.group;
+
+  defaultUser = "firefly-iii";
+  defaultGroup = "firefly-iii";
+
+  artisan = "${cfg.package}/artisan";
+
+  fileenv-func = ''
+    file_env() {
+      local var="$1"
+      local fileVar="''${var}_FILE"
+      local def="''${2:-}"
+      if [ "''${!var:-}" ] && [ "''${!fileVar:-}" ]; then
+        echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+      exit 1
+      fi
+      local val="$def"
+      if [ "''${!var:-}" ]; then
+        val="''${!var}"
+      elif [ "''${!fileVar:-}" ]; then
+        val="$(< "''${!fileVar}")"
+      fi
+      printf "$var=\"$val\"\n" >> /tmp/firefly-iii-env.conf
+      export "$var"="$val"
+      unset "$fileVar"
+    }
+
+    envs=(
+      SITE_OWNER
+      APP_KEY
+      DB_CONNECTION
+      DB_HOST
+      DB_PORT
+      DB_DATABASE
+      DB_USERNAME
+      DB_PASSWORD
+      PGSQL_SSL_MODE
+      PGSQL_SSL_ROOT_CERT
+      PGSQL_SSL_CERT
+      PGSQL_SSL_KEY
+      PGSQL_SSL_CRL_FILE
+      REDIS_HOST
+      REDIS_PASSWORD
+      REDIS_PORT
+      COOKIE_DOMAIN
+      MAIL_DRIVER
+      MAIL_HOST
+      MAIL_PORT
+      MAIL_FROM
+      MAIL_USERNAME
+      MAIL_PASSWORD
+      MAIL_ENCRYPTION
+      MAILGUN_DOMAIN
+      MAILGUN_SECRET
+      MAILGUN_ENDPOINT
+      MANDRILL_SECRET
+      SPARKPOST_SECRET
+      MAPBOX_API_KEY
+      FIXER_API_KEY
+      LOGIN_PROVIDER
+      ADLDAP_CONNECTION_SCHEME
+      ADLDAP_CONTROLLERS
+      ADLDAP_PORT
+      ADLDAP_BASEDN
+      ADLDAP_ADMIN_USERNAME
+      ADLDAP_ADMIN_PASSWORD
+      ADLDAP_ACCOUNT_PREFIX
+      ADLDAP_ACCOUNT_SUFFIX
+      WINDOWS_SSO_ENABLED
+      WINDOWS_SSO_DISCOVER
+      WINDOWS_SSO_KEY
+      ADLDAP_SYNC_FIELD
+      TRACKER_SITE_ID
+      TRACKER_URL
+    )
+
+    for e in "''${envs[@]}"; do
+      file_env "$e"
+    done
+  '';
+
+  settingsFormat = pkgs.formats.keyValue {
+    mkKeyValue = lib.flip lib.generators.mkKeyValueDefault "=" {
+      mkValueString = v: with builtins;
+        if isInt             v then toString v
+        else if isString     v then "\"${v}\""
+        else if true  ==     v then "True"
```

Comment:
```suggestion
        else if isBool       v lib.boolToString v
```

Reply:
I've removed this completely. Turns out if you use enough nix you don't need either the env file or the huge function from Wordpress.